src/presentation_exchange.rs

```diff
@@ -1,80 +1,772 @@
+    pub fn set_name(mut self, name: String) -> Self {
+        self.name = Some(name);
+        self
+    }
+
+    /// Return the name of the presentation definition.
+    pub fn name(&self) -> Option<&String> {
+        self.name.as_ref()
+    }
+
+    /// Set the purpose of the presentation definition.
+    ///
+    /// The [PresentationDefinition] MAY contain a purpose property. If present, its value MUST be a string that
+    /// describes the purpose for which the Presentation Definition's inputs are being used for.
+    pub fn set_purpose(mut self, purpose: String) -> Self {
+        self.purpose = Some(purpose);
+        self
+    }
+
+    /// Return the purpose of the presentation definition.
+    pub fn purpose(&self) -> Option<&String> {
+        self.purpose.as_ref()
+    }
+
+    /// Attach a format to the presentation definition.
+    ///
+    /// The Presentation Definition MAY include a format property. If present,
+    /// the value MUST be an object with one or more properties matching the
+    /// registered Claim Format Designations (e.g., jwt, jwt_vc, jwt_vp, etc.).
+    ///
+    /// The properties inform the [Holder](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:holder) of the Claim format configurations the [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) can process.
+    /// The value for each claim format property MUST be an object composed as follows:
+    ///
+    /// The object MUST include a format-specific property (i.e., alg, proof_type)
+    /// that expresses which algorithms the [Verifier](https://identity.foundation/presentation-exchange/spec/v2.0.0/#term:verifier) supports for the format.
+    /// Its value MUST be an array of one or more format-specific algorithmic identifier references,
+    /// as noted in the Claim Format Designations section.
+    ///
+    /// See: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition](https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition)
+    pub fn set_format(mut self, format: ClaimFormat) -> Self {
+        self.format = Some(format);
+        self
+    }
+
+    /// Return the format of the presentation definition.
+    pub fn format(&self) -> Option<&ClaimFormat> {
+        self.format.as_ref()
+    }
+
+    /// Validate a presentation submission against the presentation definition.
+    ///
+    /// Checks the underlying presentation submission parsed from the authorization response,
+    /// against the input descriptors of the presentation definition.
+    pub fn validate_authorization_response(
+        &self,
+        auth_response: &AuthorizationResponse,
+    ) -> Result<()> {
+        match auth_response {
+            AuthorizationResponse::Jwt(jwt) => {
+                bail!("Authorization Response Presentation Definition Validation Not Implemented.")
+            }
```

Comment:
TODO: Handle Jwt encoded authorized response.

Reply:
Should we resolve this now or in another PR?

Reply:
Another PR